[BUG] Setting certain documents as linked document causes internal server error

### Description

Setting certain documents as a linked document (custom field) seems to cause an internal server error.
The same pdf files don't cause the issue on another instance of paperless-ngx, which leads me to believe it has something to do with certain custom fields or the UI, rather than the file itself?

### Steps to reproduce

1. Add custom field of type "Document link" to a document 
2. Add certain document in that field.
3. Click save

### Webserver logs

```bash
paperless_webserver_1  | [2025-01-03 13:15:40,409] [ERROR] [django.request] Internal Server Error: /api/documents/80/
paperless_webserver_1  | Traceback (most recent call last):
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 518, in thread_handler
paperless_webserver_1  |     raise exc_info[1]
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
paperless_webserver_1  |     response = await get_response(request)
paperless_webserver_1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 518, in thread_handler
paperless_webserver_1  |     raise exc_info[1]
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
paperless_webserver_1  |     response = await wrapped_callback(
paperless_webserver_1  |                ^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 468, in __call__
paperless_webserver_1  |     ret = await asyncio.shield(exec_coro)
paperless_webserver_1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
paperless_webserver_1  |     result = self.fn(*self.args, **self.kwargs)
paperless_webserver_1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 522, in thread_handler
paperless_webserver_1  |     return func(*args, **kwargs)
paperless_webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
paperless_webserver_1  |     return view_func(request, *args, **kwargs)
paperless_webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/viewsets.py", line 124, in view
paperless_webserver_1  |     return self.dispatch(request, *args, **kwargs)
paperless_webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 509, in dispatch
paperless_webserver_1  |     response = self.handle_exception(exc)
paperless_webserver_1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 469, in handle_exception
paperless_webserver_1  |     self.raise_uncaught_exception(exc)
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
paperless_webserver_1  |     raise exc
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 506, in dispatch
paperless_webserver_1  |     response = handler(request, *args, **kwargs)
paperless_webserver_1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/src/paperless/src/documents/views.py", line 393, in update
paperless_webserver_1  |     response = super().update(request, *args, **kwargs)
paperless_webserver_1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/mixins.py", line 68, in update
paperless_webserver_1  |     self.perform_update(serializer)
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/mixins.py", line 78, in perform_update
paperless_webserver_1  |     serializer.save()
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/drf_writable_nested/mixins.py", line 233, in save
paperless_webserver_1  |     return super(BaseNestedModelSerializer, self).save(**kwargs)
paperless_webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/serializers.py", line 203, in save
paperless_webserver_1  |     self.instance = self.update(self.instance, validated_data)
paperless_webserver_1  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/src/paperless/src/documents/serialisers.py", line 881, in update
paperless_webserver_1  |     super().update(instance, validated_data)
paperless_webserver_1  |   File "/usr/src/paperless/src/documents/serialisers.py", line 332, in update
paperless_webserver_1  |     return super().update(instance, validated_data)
paperless_webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/drf_writable_nested/mixins.py", line 290, in update
paperless_webserver_1  |     self.update_or_create_reverse_relations(instance, reverse_relations)
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/drf_writable_nested/mixins.py", line 188, in update_or_create_reverse_relations
paperless_webserver_1  |     related_instance = serializer.save(**save_kwargs)
paperless_webserver_1  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/serializers.py", line 208, in save
paperless_webserver_1  |     self.instance = self.create(validated_data)
paperless_webserver_1  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  |   File "/usr/src/paperless/src/documents/serialisers.py", line 600, in create
paperless_webserver_1  |     self.reflect_doclinks(document, custom_field, validated_data["value"])
paperless_webserver_1  |   File "/usr/src/paperless/src/documents/serialisers.py", line 716, in reflect_doclinks
paperless_webserver_1  |     elif document.id not in target_doc_field_instance.value:
paperless_webserver_1  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
paperless_webserver_1  | TypeError: 'in <string>' requires string as left operand, not int
paperless_webserver_1  | [2025-01-03 13:15:43,015] [WARNING] [django.request] Bad Request: /api/documents/80/
```


### Browser logs

```bash
Error popup in the webui:
{"headers":{"normalizedNames":{},"lazyUpdate":null},"status":500,"statusText":"Internal Server Error","url":"http://192.168.1.2:8000/api/documents/80/","ok":false,"name":"HttpErrorResponse","message":"Http failure response for http://192.168.1.2:8000/api/documents/80/: 500 Internal Server Error","error":"\n<!doctype html>\n<html lang=\"en\">\n<head>\n  <title>Server Error (500)</title>\n</head>\n<body>\n  <h1>Server Error (500)</h1><p></p>\n</body>\n</html>\n"}
```


### Paperless-ngx version

2.13.5

### Host OS

Debian 12

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.13.5",
    "server_os": "Linux-6.1.0-25-amd64-x86_64-with-glibc2.36",
    "install_type": "docker",
    "storage": {
        "total": 104348344320,
        "available": 35895164928
    },
    "database": {
        "type": "postgresql",
        "url": "paperless",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0011_remove_mailrule_assign_tag_squashed_0024_alter_mailrule_name_and_more",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://broker:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "index_status": "OK",
        "index_last_modified": "2025-01-03T13:44:34.479716+01:00",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2025-01-03T12:05:00.048963Z",
        "classifier_error": null
    }
}
```


### Browser

Chrome, Safari

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/documents/serialisers.py
+++ b/src/documents/serialisers.py
@@ -668,6 +668,10 @@ class CustomFieldInstanceSerializer(serializers.ModelSerializer):
                         f"Value must be an id of an element in {select_options}",
                     )
             elif field.data_type == CustomField.FieldDataType.DOCUMENTLINK:
+                if not (isinstance(data["value"], list) or data["value"] is None):
+                    raise serializers.ValidationError(
+                        "Value must be a list",
+                    )
                 doc_ids = data["value"]
                 if Document.objects.filter(id__in=doc_ids).count() != len(
                     data["value"],

--- a/src/documents/tests/test_api_custom_fields.py
+++ b/src/documents/tests/test_api_custom_fields.py
@@ -886,6 +886,7 @@ class TestCustomFieldsAPI(DirectoriesMixin, APITestCase):
             - Document & custom field exist
         WHEN:
             - API request to set a field value to a document that does not exist
+            - API request to set a field value to empty string
         THEN:
             - HTTP 400 is returned
             - No field instance is created or attached to the document
@@ -915,6 +916,19 @@ class TestCustomFieldsAPI(DirectoriesMixin, APITestCase):
         self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertEqual(CustomFieldInstance.objects.count(), 0)
         self.assertEqual(len(doc.custom_fields.all()), 0)
+
+        resp = self.client.patch(
+            f"/api/documents/{doc.id}/",
+            data={
+                "custom_fields": [
+                    {"field": custom_field_documentlink.id, "value": ""},
+                ],
+            },
+            format="json",
+        )
+
+        self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(CustomFieldInstance.objects.count(), 0)
 
     def test_custom_field_not_null(self):
         """

